chapter_2/component1/CMakeLists.txt

```diff
@@ -3,23 +3,39 @@
 # SPDX-License-Identifier: MIT
 
 project(
-  ch2.framework.component1
-  VERSION 1.0
-  DESCRIPTION "Chapter 2 component 1 implementation"
-  LANGUAGES CXX)
+    ch2.framework.component1
+    VERSION 1.0
+    DESCRIPTION "Chapter 2 component 1 implementation"
+    LANGUAGES CXX
+)
 
 add_library(
-  ${PROJECT_NAME} STATIC) # Create a static library target named
-                          # ch2.framework.component1 (retrieved from project
-                          # name variable)
+    ${PROJECT_NAME} STATIC
```

Comment:
Personally I'm really not fond of using `${PROJECT_NAME}` for the "main" target, because it just obscures its name. How often does a project's name change that this is warranted?

Reply:
I think this stems mostly from the fact that lots of people just copy an existing CMakeLists.txt when starting a new project and does not add much value afterward. 

Considering the book, the practice is widely used so we should mention it, but probably not encouraging it. What do you think @mustafakemalgilor ?

Reply:
I find specifying the target name each time is very tedious. Personally, I often create macros/functions for creating a target to prevent code repetition (click [here](https://github.com/nettsi/hadouken/blob/master/cmake/modules/core/MakeTarget.cmake#L33 ) for an example). I also tend to use cascaded project names to automatically name sub-components after the parent names. About the obscurity, the people who use CMake are accustomed to simple concepts like variables and such. Thus, I don't think this is confusing or obscure for our target audience.

@bernedom I think this is not a thing to discourage people away from. Discouraging requires side effects as a backing rationale. It is a personal choice and there is nothing bad about using a scoped variable (e.g. PROJECT_NAME) for your target name if you want to structure your project like that. 

To conclude, the things I've mentioned above are my personal opinions about the topic. You know the saying about opinions, everybody has one. Since there is a majority on disagreement, I'll substitute them with the regular names, for the present and the future examples.

Reply:
In bigger projects, things indeed may work a bit different. The PDF I received about the book describes the target audience to be very wide. I think since an upcoming chapter will cover custom CMake modules and functionality, maybe this sort of abstraction could be deferred to that instead with a preposition that it could be more useful for large systems. I reckon that while we are in the shallow waters, such abstractions could be avoided in the spirit of KISS.

Reply:
Replaced all ${PROJECT_NAME} usage in chapter 2 content.